router.go

```diff
@@ -604,22 +606,23 @@ func (r *Router) Find(method, path string, c Context) {
 		return // nothing matched at all
 	}
 
-	if matchedHandler != nil {
-		ctx.handler = matchedHandler
+	if matchedRouteMethod != nil {
+		ctx.handler = matchedRouteMethod.handler
+		ctx.path = matchedRouteMethod.ppath
+		ctx.pnames = matchedRouteMethod.pnames
 	} else {
 		// use previous match as basis. although we have no matching handler we have path match.
 		// so we can send http.StatusMethodNotAllowed (405) instead of http.StatusNotFound (404)
 		currentNode = previousBestMatchNode
 
+		ctx.path = path
```

Comment:
See comment below. This is incorrect as `path` must be path of the route not path of the request.

Reply:
Fixed